WooCommerce/Classes/POS/Presentation/POSFloatingControlView.swift

```diff
@@ -32,22 +34,40 @@ struct POSFloatingControlView: View {
                 HStack {
                     Text("⋯")
                         .font(Constants.ellipsisFont)
+                        .foregroundStyle(fontColor)
                 }
                 .frame(width: Constants.size, height: Constants.size)
             }
-            .background(Color.white)
+            .background(backgroundColor)
             .cornerRadius(Constants.cornerRadius)
             .disabled(viewModel.isExitPOSDisabled)
             HStack {
-                CardReaderConnectionStatusView(connectionViewModel: viewModel.cardReaderConnectionViewModel)
+                CardReaderConnectionStatusView(connectionViewModel: viewModel.cardReaderConnectionViewModel, totalsViewModel: totalsViewModel)
                     .padding(Constants.cardStatusPadding)
+                    .foregroundStyle(fontColor)
             }
             .frame(height: Constants.size)
-            .background(Color.white)
+            .background(backgroundColor)
             .cornerRadius(Constants.cornerRadius)
         }
         .background(Color.clear)
     }
+
+    private var backgroundColor: Color {
+        if totalsViewModel.paymentState == .processingPayment {
+            return Color(.wooCommercePurple(.shade80))
+        } else {
+            return .white
+        }
+    }
+
+    private var fontColor: Color {
+        if totalsViewModel.paymentState == .processingPayment {
+            return .posSecondaryTextDark
+        } else {
+            return .primaryText
+        }
+    }
```

Comment:
Again, it feels like unwanted coupling to have this view know about the `totalsViewModel`